### DISCUSSION TITLE
Tinygo > 0.31 has breaking change

### DISCUSSION THREAD

**Discussion by mlathara** [aquasecurity/trivy#7753] [Ideas]
### Description

This isn't really a feature request per se -- I just wanted to share some incompatibilities I ran into when trying to create an [external module](https://aquasecurity.github.io/trivy/v0.56/docs/advanced/modules/). When using tinygo version 0.33 (latest), I ran into errors when attempting to build the wasm binary. 

```
../../go/pkg/mod/github.com/aquasecurity/trivy@v0.56.2/pkg/module/wasm/sdk.go:146:10: cannot use uintptr(size) (value of type uintptr) as int value in assignment
../../go/pkg/mod/github.com/aquasecurity/trivy@v0.56.2/pkg/module/wasm/sdk.go:147:10: cannot use uintptr(size) (value of type uintptr) as int value in assignment
../../go/pkg/mod/github.com/aquasecurity/trivy@v0.56.2/pkg/module/wasm/sdk.go:163:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
../../go/pkg/mod/github.com/aquasecurity/trivy@v0.56.2/pkg/module/wasm/sdk.go:164:9: cannot use uintptr(size) (value of type uintptr) as int value in struct literal
```

After a bit of digging, I found that tinygo [updated its reflect](https://github.com/tinygo-org/tinygo/pull/4156) package, which causes the breakage ([example](https://github.com/tinygo-org/tinygo/issues/4318)). 

So - just wanted to call out that trivy will also need some minor updates here if/when you decide to update tinygo > 0.31 and/or update Go > 1.22

edit: adding a link for Tinygo's recommendations on this issue https://tinygo.org/docs/guides/compatibility/#reflectsliceheader-and-reflectstringheader

### Target

None

### Scanner

None

**Comment by nikpivkin** [comment]
Duplicate of https://github.com/aquasecurity/trivy/discussions/7568

**Reply by michaeljsaenz** [reply]
is there a temp workaround to get `mage test:unit` passing?
